src/ai/backend/manager/api/container_registry.py

```diff
@@ -76,13 +78,89 @@ async def disassociate_with_group(request: web.Request, params: Any) -> web.Resp
     return web.json_response({})
 
 
+@server_status_required(ALL_ALLOWED)
+@check_api_params(
+    t.Dict({
+        "type": t.String,
+        "event_data": t.Dict({
+            "resources": t.List(
+                t.Dict({
+                    "resource_url": t.String,
+                    "tag": t.String,
+                }).allow_extra("*")
+            ),
+            "repository": t.Dict({
+                "namespace": t.String,
+                "name": t.String,
+            }).allow_extra("*"),
+        }).allow_extra("*"),
+    }).allow_extra("*")
+)
+async def harbor_webhook_handler(request: web.Request, params: Any) -> web.Response:
+    auth_header = request.headers.get("Authorization", None)
+    event_type = params["type"]
+    resources = params["event_data"]["resources"]
+    project = params["event_data"]["repository"]["namespace"]
+    img_name = params["event_data"]["repository"]["name"]
+    log.info("HARBOR_WEBHOOK_HANDLER (event_type:{})", event_type)
+
+    root_ctx: RootContext = request.app["_root.context"]
+    async with root_ctx.db.begin_session() as db_sess:
+        for resource in resources:
+            resource_url = resource["resource_url"]
+            registry_url = resource_url.split("/")[0]
+
+            query = sa.select(ContainerRegistryRow).where(
+                (ContainerRegistryRow.type == ContainerRegistryType.HARBOR2)
+                & (ContainerRegistryRow.url.like(f"%{registry_url}%"))
+                & (ContainerRegistryRow.project == project)
+            )
+            registry_row = (await db_sess.execute(query)).scalars().one_or_none()
+
+            if not registry_row:
+                log.error(
+                    "Harbor container registry row not found! (registry url: {}, project: {})",
+                    registry_url,
+                    project,
+                )
+                return web.json_response({}, status=500)
+
+            if auth_header:
+                if (
+                    not registry_row.extra
+                    or registry_row.extra.get("webhook_auth_header", None) != auth_header
+                ):
+                    log.warning("Unauthorized request from Harbor webhook")
+                    return web.json_response({}, status=401)
```

Comment:
When a DB SCAN occurs for an unauthenticated request, it looks like a 500 error may occur before an authentication error.

Reply:
It's true.
However, in the current structure, the harbor webhook auth value is stored in the `extra` column of `ContainerRegistryRow`, so the auth verification can only be performed after the row has been retrieved.